ra/ra.go

```diff
@@ -554,20 +554,29 @@ func (ra *RegistrationAuthorityImpl) NewAuthorization(ctx context.Context, reque
 	if err != nil {
 		return core.Authorization{}, err
 	}
+
+	if features.Enabled(features.NewAuthorizationSchema) {
+		resultPB, err := ra.SA.NewAuthorization(ctx, authzPB)
+		if err != nil {
+			return core.Authorization{}, err
+		}
+		id := fmt.Sprintf("%d", *resultPB.Id)
```

Comment:
I understand why this type conversion is happening based on our discussions but I think if I were reading it with just the code context I wouldn't fully grok the "why". Can you add a brief comment?